src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -114,14 +115,34 @@ public void setProjectId(String projectId) {
     this.projectId = projectId;
   }
 
+  @Deprecated
   public String getZone() {
-    return this.zone;
+    return getLocation();
   }
 
+  @Deprecated
   @DataBoundSetter
   public void setZone(String zone) {
-    Preconditions.checkArgument(!Strings.isNullOrEmpty(zone));
-    this.zone = zone;
+    setLocation(zone);
+  }
+
+  public String getLocation() {
+    setupLocation();
```

Comment:
If setZone now calls setLocation directly, I'm not sure setupLocation() is needed anymore.

Reply:
This is needed as an intermediate step to preserve compatibility across versions. Unless we have this, then location won't be set properly when loading from an old configuration that only has zone set.

Reply:
That makes sense. Thanks for clarifying.